prov/efa/src/rdm/rxr_srx.c

```diff
@@ -0,0 +1,188 @@
+/*
+ * Copyright (c) Amazon.com, Inc. or its affiliates.
+ * All rights reserved.
+ *
+ * This software is available to you under a choice of one of two
+ * licenses.  You may choose to be licensed under the terms of the GNU
+ * General Public License (GPL) Version 2, available from the file
+ * COPYING in the main directory of this source tree, or the
+ * BSD license below:
+ *
+ *     Redistribution and use in source and binary forms, with or
+ *     without modification, are permitted provided that the following
+ *     conditions are met:
+ *
+ *      - Redistributions of source code must retain the above
+ *        copyright notice, this list of conditions and the following
+ *        disclaimer.
+ *
+ *      - Redistributions in binary form must reproduce the above
+ *        copyright notice, this list of conditions and the following
+ *        disclaimer in the documentation and/or other materials
+ *        provided with the distribution.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS
+ * BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN
+ * ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+#include "efa.h"
+#include "rxr_srx.h"
+
+/**
+ * @brief This call is invoked by the peer provider to obtain the receive
+ * buffers where an incoming message should be placed.
+ *
+ * @param[in] srx the fid_peer_srx
+ * @param[in] addr the source address of the incoming message
+ * @param[in] size the size of the incoming message (ignored unless multi-receive)
+ * @param[out] rx_entry the obtained receive buffer
+ * @return int 0 on success, a negative integer on failure
+ */
+static int rxr_srx_get_msg(struct fid_peer_srx *srx, fi_addr_t addr,
+		       size_t size, struct fi_peer_rx_entry **rx_entry)
+{
+    return -FI_ENOSYS;
+}
+
+/**
+ * @brief This call is invoked by the peer provider to obtain the receive
+ * buffers where an incoming tagged message should be placed.
+ *
+ * @param[in] srx the fid_peer_srx
+ * @param[in] addr the source address of the incoming message
+ * @param[in] size the size of the incoming message (ignored unless multi-receive)
+ * @param[out] rx_entry the obtained receive buffer
+ * @return int 0 on success, a negative integer on failure
+ */
+static int rxr_srx_get_tag(struct fid_peer_srx *srx, fi_addr_t addr,
```

Comment:
`*_tag` is ambiguous IMO, suggest being a verbose name, i.e. `*_tagged_msg`

Reply:
I may change it to `*_tag_entry` to be consistent with the API name. But I agree with you that this API name is kind of not perfect.

Reply:
No, the man page is wrong ... the API name is exactly `get_tag`. I will keep the current `get_tag` to be consistent.